### DISCUSSION TITLE
lozenge symbol is drawn with awkward spacing

### DISCUSSION THREAD

**Discussion by axodentally** [typst/typst#4472] [General]
### Description

Given the following code:
```latex
$square B lozenge B$
```
Typst produces:
![image](https://github.com/typst/typst/assets/24368475/ca0f978e-ab67-4ce2-80b5-5c2b2db2bab1)

While I would expect it to have equal spacing like in LaTeX:
![image](https://github.com/typst/typst/assets/24368475/27d2f519-d993-4ffa-908d-5907e5257ee0)

with the latex code:
```latex
$\square B \lozenge B$
```

Furthermore I also find the size and positioning of the symbols look a little "off" and better in latex, but I that could just be my personal preference.

### Operating system

Web app

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enivex** [comment]
For whatever reason amssymb uses "normal" (`\mathord`) for U+25CA, but [unicode specifies](https://www.unicode.org/Public/math/revision-15/MathClass-15.txt) that it should be "binary". This is why the spacing is different, but you can force a different class with `class()`.

As for size and position, I suspect you're not using the exact same font through luatex/xetex and unicode-math. I also believe amssymb uses its own font for the symbols it defines, so you'd need `\mdlgwhtlozenge` (see [the list](https://ctan.uib.no/macros/unicodetex/latex/unicode-math/unimath-symbols.pdf))

**Reply by axodentally** [reply]
Okay, that would explain that typst draws it differently, in that case it seems to be the right choice.

But is the spacing also supposed to be like that? To me it looks a little odd in mathematical formulas and such. Does the unicode spec define such spacing?

**Reply by Enivex** [reply]
> Okay, that would explain that typst draws it differently, in that case it seems to be the right choice.
> 
> But is the spacing also supposed to be like that? To me it looks a little odd in mathematical formulas and such. Does the unicode spec define such spacing?

Spacing between two symbols is determined by their classes, and the unicode spec says what class each symbol belongs to.

But like I said, you can change the class to your liking in typst using the `class` function.

**Reply by SkiFire13** [reply]
> But is the spacing also supposed to be like that? To me it looks a little odd in mathematical formulas and such.

It's the same spacing used for other binary operators, like for example `+`. Latex also has spacing around the other binary operators.

**Reply by axodentally** [reply]
alright, got it. Thanks for the explanation, makes sense now :)
